presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -312,4 +324,21 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce query scan physical bytes limits
+     */
+    private void enforceScanLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            BasicQueryStats basicQueryStats = query.getBasicQueryInfo().getQueryStats();
+            QueryStats queryStats = query.getQueryInfo().getQueryStats();
+            DataSize scan = query.getBasicQueryInfo().getQueryStats().getRawInputDataSize();
```

Comment:
This is still a bit expensive since getting querystats involves -> for all stages, for all tasks, for all metrics.

The way CPU does it is still cheaper -> all stages, all tasks, just the cpu - so we can do something similar here and make it even cheaper - 

We can follow a similar flow here

SqlQueryManager -> SqlQuerySchedulerInterface#getTotalCpuTime -> SqlStageExecution#getTotalCpuTime

and just sum up the bytes involved.

Reply:
I'm not certain if your approach about summing up bytes from `SqlStageExecution#getTotalScanBytes` is accurate... it's always the best practice to inherit scanned byte size from existing querystats to ensure the result is correct. I suggest we maintain current implementation with `SqlQueryManager:getBasicQueryInfo-> BasicQueryStats:getQueryStats` . How much cheaper are we suggesting to reduce if we want to bypass the existing logic with a risk that we might have a false result? 
My option is that `rawInputDataSize` should be the only reliable source of truth when setting against with the hard scan limit.

Reply:
I don't agree that `BasicQueryStats:getQueryStats` is the only trusted source. As you can see, we are already doing this for total cpu ms and memory limits so I don't see why raw input bytes will be any different here. Again, my only concern is that this is a single thread already doing cpu and memory enforcements and we should keep it as light weight as possible and by keeping this trend, we will ensure that the new entries that are added also follow this lighter weight approach.

Reply:
understood, thank you! I have implemented your approach, I think it is safer and cheaper to acquire actual number via `SqlQuerySchedulerInterface::getRawInputDataSize` -> `QueryExecution::getRawInputDataSize`, and sum up all task's stats via `SqlStageExecution::getRawInputDataSize`. Please review the revised commit again, many thanks!